### PR TITLE
Add storage operations interface which will be implemented by providers like AWS and GCE

### PR DESCRIPTION
**Pull request by harsh-px**
This is basically a generic version of storage ops interface in the aws driver: https://github.com/libopenstorage/openstorage/blob/master/volume/drivers/aws/aws_storage.go#L44

### CODE REVIEW

File: pkg/storageops/storageops.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package storageops
+
+const (
+	// SetIdentifierNone is an default identifier to group all disks managed by
+	// storage ops
```
Comment (adityadani) [inline pkg/storageops/storageops.go]: nit: is a default identifier to group all disks from a particular set

File: pkg/storageops/storageops.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	GetDeviceID(template interface{}) string
+	// Attach volumeID.
+	// Return attach path.
+	Attach(volumeID string) (string, error)
+	// Detach volumeID.
+	Detach(volumeID string) error
+	// Delete volumeID.
+	Delete(volumeID string) error
+	// FreeDevices returns free block devices on the instance
+	FreeDevices(blockDeviceMappings []interface{}, rootDeviceName string) ([]string, error)
```
Comment (adityadani) [inline pkg/storageops/storageops.go]: nit: add what blockDeviceMappings interface is

File: pkg/storageops/storageops.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+const (
+	// SetIdentifierNone is an default identifier to group all disks managed by
+	// storage ops
+	SetIdentifierNone = "None"
+)
+
+// Custom storage operation error codes.
+const (
+	_ = iota + 5000
+	ErrVolDetached
```
Comment (adityadani) [inline pkg/storageops/storageops.go]: nit: doc for each constant.

File: pkg/storageops/storageops.go (hunk 2 above)
Comment (harsh-px) [reply pkg/storageops/storageops.go]: done

File: pkg/storageops/storageops.go (hunk 3 above)
Comment (harsh-px) [reply pkg/storageops/storageops.go]: done

File: pkg/storageops/storageops.go (hunk 1 above)
Comment (harsh-px) [reply pkg/storageops/storageops.go]: done

**Comment by harsh-px** [comment]
@lpabon : This PR is approved and I will merge this in now. Feel free to post your comments and I will resolve them in subsequent commits.

**Pull request merged**